### ISSUE TITLE
Sync failures after upgrade from 0.0.16 to 0.0.19

### ISSUE THREAD

**Issue by imagebastler** [redhat-cop/group-sync-operator#204]
Hi all

We upgraded group-sync-operator from 0.0.16 to 0.0.19 on an OpenShift 4.10 cluster using azure ad as provider. Using this version we get the following warning in group-sync manager container:

```
2022-07-08T11:31:21.460Z INFO controllers.GroupSync Beginning Sync {"groupsync": "group-sync-operator/xxx-xxx-team-groupsync", "Provider": "xxx-azure-sso"}
2022-07-08T11:33:20.452Z INFO syncer_azure Warning: Username for user cannot be found in Group ID 'e872bae1-6e8e-4a68-aade-a3b073c46b3f'
2022-07-08T11:33:20.452Z INFO syncer_azure Warning: Username for user cannot be found in Group ID 'e872bae1-6e8e-4a68-aade-a3b073c46b3f'
2022-07-08T11:33:20.452Z INFO syncer_azure Warning: Username for user cannot be found in Group ID 'e872bae1-6e8e-4a68-aade-a3b073c46b3f'
2022-07-08T11:33:20.452Z INFO syncer_azure Warning: Username for user cannot be found in Group ID 'e872bae1-6e8e-4a68-aade-a3b073c46b3f'
2022-07-08T11:33:20.452Z INFO syncer_azure Warning: Username for user cannot be found in Group ID 'e872bae1-6e8e-4a68-aade-a3b073c46b3f'
2022-07-08T11:33:20.452Z INFO syncer_azure Warning: Username for user cannot be found in Group ID 'e872bae1-6e8e-4a68-aade-a3b073c46b3f'
2022-07-08T11:33:20.452Z INFO syncer_azure Warning: Username for user cannot be found in Group ID 'e872bae1-6e8e-4a68-aade-a3b073c46b3f'
2022-07-08T11:33:20.452Z INFO syncer_azure Warning: Username for user cannot be found in Group ID 'e872bae1-6e8e-4a68-aade-a3b073c46b3f'
2022-07-08T11:33:20.587Z INFO controllers.GroupSync Sync Completed Successfully {"groupsync": "group-sync-operator/xxx-xxx-team-groupsync", "Provider": "xxx-azure-sso", "Groups Created or Updated": 1, "Groups Pruned": 0}
```
After one sync run all synced groups are empty -> all members are removed.
```
$ oc get group
NAME                                          USERS
DG_XXX_XXX_TEAM
```

Group sync definition
```
apiVersion: redhatcop.redhat.io/v1alpha1
kind: GroupSync
metadata:
  name: xxx-xxx-team-groupsync
  namespace: group-sync-operator
spec:
  providers:
  - azure:
      credentialsSecret:
        kind: Secret
        name: azure-group-sync
        namespace: group-sync-operator
      groups:
      - XXX_CON_TEAM
      userNameAttributes:
      - onPremisesSamAccountName
    name: xxx-azure-sso
  schedule: '*/30 * * * *'
```

We are sure that "onPremisesSamAccountName" is correct since this works for 0.0.16. And as soon as we go back to 0.0.16 everything works as expected again.

Thanks for any hints.

Regards,
Roger

**Comment by sabre1041** [comment]
Hi there!

There was a substantial library change that did occur between those two versions. Testing was performed to ensure that it was able to pull in metadata from another field. Is there any additional information on the group structure or any additional details that you can provide, it would be helpful

**Comment by jolbax** [comment]
Hi @sabre1041 
We were analysing this issue and it looks like that "[isUsernamePresent](https://github.com/redhat-cop/group-sync-operator/blob/bc3663076cb7ca20da210a3f3b99b25fed48e33c/pkg/syncer/azure.go#L357)" works only if one is looking for a attributes returned by default and not on select

**Comment by jolbax** [comment]
Hi @sabre1041 
I created a PR https://github.com/redhat-cop/group-sync-operator/pull/205

**Comment by sabre1041** [comment]
Resolved via #205
